networkx/algorithms/tests/test_graph_hashing.py

```diff
@@ -250,7 +251,148 @@ def test_digest_size():
+    Gb_hash = nx.weisfeiler_lehman_graph_hash(Gb)
+    assert Ga_hash != Gb_hash
+
+    Tree1 = nx.DiGraph()
+    Tree1.add_edges_from([(0, 4), (1, 5), (2, 6), (3, 7)])
+    Tree1.add_edges_from([(4, 8), (5, 8), (6, 9), (7, 9)])
+    Tree1.add_edges_from([(8, 10), (9, 10)])
+    nx.set_node_attributes(
+        Tree1, {10: "s", 8: "a", 9: "a", 4: "b", 5: "b", 6: "b", 7: "b"}, "weight"
+    )
+    Tree2 = copy.deepcopy(Tree1)
+    nx.set_node_attributes(Tree1, {0: "d", 1: "c", 2: "d", 3: "c"}, "weight")
+    nx.set_node_attributes(Tree2, {0: "d", 1: "d", 2: "c", 3: "c"}, "weight")
+    Tree1_hash_short = nx.weisfeiler_lehman_graph_hash(
+        Tree1, iterations=1, node_attr="weight"
+    )
+    Tree2_hash_short = nx.weisfeiler_lehman_graph_hash(
+        Tree2, iterations=1, node_attr="weight"
+    )
+    assert Tree1_hash_short == Tree2_hash_short
+    Tree1_hash = nx.weisfeiler_lehman_graph_hash(
+        Tree1, node_attr="weight"
+    )  # Default is 3 iterations
+    Tree2_hash = nx.weisfeiler_lehman_graph_hash(Tree2, node_attr="weight")
+    assert Tree1_hash != Tree2_hash
+
+
+def test_trivial_labels_isomorphism():
+    """
+    Test that trivial labelling of the graph should not change isomorphism verdicts.
+    """
+    n, r = 100, 10
+    p = 1.0 / r
+    for i in range(1, r + 1):
+        G1 = nx.erdos_renyi_graph(n, p * i, seed=500 + i)
+        G2 = nx.erdos_renyi_graph(n, p * i, seed=42 + i)
+        equal = nx.weisfeiler_lehman_graph_hash(G1) == nx.weisfeiler_lehman_graph_hash(
+            G2
+        )
+        nx.set_node_attributes(G1, values=1, name="weight")
+        nx.set_node_attributes(G2, values=1, name="weight")
+        equal_node = nx.weisfeiler_lehman_graph_hash(
+            G1, node_attr="weight"
+        ) == nx.weisfeiler_lehman_graph_hash(G2, node_attr="weight")
+        nx.set_edge_attributes(G1, values="a", name="e_weight")
+        nx.set_edge_attributes(G2, values="a", name="e_weight")
+        equal_edge = nx.weisfeiler_lehman_graph_hash(
+            G1, edge_attr="e_weight"
+        ) == nx.weisfeiler_lehman_graph_hash(G2, edge_attr="e_weight")
+        equal_both = nx.weisfeiler_lehman_graph_hash(
+            G1, edge_attr="e_weight", node_attr="weight"
+        ) == nx.weisfeiler_lehman_graph_hash(
+            G2, edge_attr="e_weight", node_attr="weight"
+        )
+        assert equal == equal_node
+        assert equal_node == equal_edge
+        assert equal_edge == equal_both
+
+
+def test_trivial_labels_isomorphism_directed():
+    """
+    Test that trivial labelling of the graph should not change isomorphism verdicts on directed graphs.
+    """
+    n, r = 100, 10
+    p = 1.0 / r
+    for i in range(1, r + 1):
+        G1 = nx.erdos_renyi_graph(n, p * i, directed=True, seed=500 + i)
+        G2 = nx.erdos_renyi_graph(n, p * i, directed=True, seed=42 + i)
+        equal = nx.weisfeiler_lehman_graph_hash(G1) == nx.weisfeiler_lehman_graph_hash(
+            G2
+        )
```

Comment:
I found this hard to parse. What do you think about the following?
```suggestion
        G1_hash = nx.weisfeiler_lehman_graph_hash(G1)
        G2_hash = nx.weisfeiler_lehman_graph_hash(G2)
        equal = (G1_hash == G2_hash)
```
and similar below.

Reply:
Sure!